### PR TITLE
generalise setup tools

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

alow us to reuse in PL followup projects, remove hardcoded paths/version

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/setup_tools.py [hunk 1]
```diff
@@ -55,107 +52,10 @@ def _load_requirements(path_dir: str , file_name: str = 'requirements.txt', comm
...
-            if err.code == 404:
-                # save the `.svg`
-                url_badge = url_badge.replace('.png', '.svg')
-                _save_file(url_badge, save_path, extension='.svg', headers=headers)
-                return save_path + '.svg'
-            return ''
-
-
-def _load_long_description(path_dir: str) -> str:
+def _load_readme_description(path_dir: str, homepage: str = __homepage__, ver: str = __version__) -> str:
```
Comment (gianscarpe) [inline pytorch_lightning/setup_tools.py:55]: I'll suggest `version` as parameter name instead of `ver`, it improves readability. What do you think? Btw, nice job :)

**Review by gianscarpe — CHANGES_REQUESTED** [review]

**Review by ydcjeff — APPROVED** [review]

File: pytorch_lightning/setup_tools.py [hunk 2]
```diff
@@ -55,107 +52,10 @@ def _load_requirements(path_dir: str , file_name: str = 'requirements.txt', comm
     return reqs
 
 
-def _parse_for_badge(
```
Comment (tchaton) [inline pytorch_lightning/setup_tools.py:58]: Why did you delete those functions ?

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/setup_tools.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/setup_tools.py:58]: have not used them and the download was time-to-time failing

**Review by williamFalcon — APPROVED** [review]
